tests/sparse_set_tests.cpp

```diff
@@ -142,34 +142,60 @@ BOOST_AUTO_TEST_CASE(test_insert_pointer) {
 /**
  * serialize and deserialize
  */
-BOOST_AUTO_TEST_CASE(test_serialize_deserialize) {
-    // insert x values; delete some values; serialize set; deserialize in new set; check equal.
-    // for deserialization, test it with and without hash compatibility.
-    const std::size_t nb_values = 1000;
-    
-    
-    tsl::sparse_set<move_only_test> set;
-    for(std::size_t i = 0; i < nb_values + 40; i++) {
-        set.insert(utils::get_key<move_only_test>(i));
+BOOST_AUTO_TEST_CASE(test_serialize_deserialize_reserve) {
```

Comment:
In the `test_serialize_deserialize` you also do some erase as it was before but not here, is there a reason? The comment still says "delete some values".